### ISSUE TITLE
Add new line breaks automatically when displaying plugin description

### ISSUE THREAD

**Issue by rikkolovescats** [bombsquad-community/plugin-manager#89]
Will have to experiment a little on what's the most appropriate character/word limit for a single line. And also if it would be a good idea to factor this limit for current ui scale.

----------

Right now, some plugins have lengthy descriptions which are unreadable in a single line, such as:

<img src="https://user-images.githubusercontent.com/106954762/208247008-925136a4-287a-44de-9408-4517886a7b02.png" width="600">

https://github.com/bombsquad-community/plugin-manager/blob/3effcbfcdf77c15913bc79fdef5f48da433ee964/plugins/utilities.json#L475

----------

These would look better if they are to be broken down in multiple lines, such as:

<img src="https://user-images.githubusercontent.com/106954762/208247089-e4133fca-da1f-4c65-b368-a558f999f490.png" width="600">

But these line breaks have to be manually added currently:
https://github.com/bombsquad-community/plugin-manager/blob/3effcbfcdf77c15913bc79fdef5f48da433ee964/plugins/utilities.json#L513

Would be cool if this could be done automatically by the plugin manager ui.

**Comment by kingsamurai123** [comment]
The below code takes the utilities.json and replace the description with '\n' for the nearest ' ' (space) within specified character limit. But it ended up doing partitions on irregular intervals. Not sure where the error exists. Is this helpful in any manner?
```
import json

character_limit = 40
filename = 'utilities.json'

def replaceNearestSpaceChar(str,charIndex):
    str = list(str)
    while (True):
        if str[charIndex] == " ":
            str[charIndex] = "\n"
            return ''.join(str),charIndex
        else:
            charIndex -= 1

with open(filename, 'r',encoding="utf-8") as f:
  data = json.load(f)

plugins_data = data['plugins']
for plugin in plugins_data:
    desc = plugins_data[plugin]['description']
    if (len(desc) > character_limit):
        index = 0
        new_desc = ''
        while index+character_limit < len(desc):
            new_desc,index = replaceNearestSpaceChar(desc,index+character_limit)
        plugins_data[plugin]['description'] = new_desc

with open(filename, 'w',encoding="utf-8") as f:
    json.dump(data,f,ensure_ascii=False, indent=4)
f.close

```

**Comment by rikkolovescats** [comment]
i think the problem seems to be that you're always passing the original `desc` to `replaceNearestSpaceChar` func, so only the final replacement of whitespace to new line character seems to persist

try replacing the relevant part in your code with this:
```diff
23c23
<         new_desc = ''
---
>         new_desc = desc
25c25
<             new_desc,index = replaceNearestSpaceChar(desc,index+character_limit)
---
>             new_desc,index = replaceNearestSpaceChar(new_desc,index+character_limit)

```

i also wonder what might be better - splitting whitespace just before the limit or just after the limit?

this one splits just after the limit:
```python
long_string = "Hello! I'm a reaaaalllyy loooong string. Split us each into our homes please!!"


def partition(string, minimum_character_offset=40):
    string_length = len(string)

    partitioned_string = ""
    partitioned_string_length = len(partitioned_string)

    while partitioned_string_length != string_length:
        next_empty_space = string[partitioned_string_length + minimum_character_offset:].find(" ")
        next_word_end_position = partitioned_string_length + minimum_character_offset + max(0, next_empty_space)
        partitioned_string += string[partitioned_string_length:next_word_end_position]
        if next_empty_space != -1:
            # Insert a line break here, there's still more partitioning to do.
            partitioned_string += "\n"
        partitioned_string_length = len(partitioned_string)

    return partitioned_string


# Make sure there are no trailing whitespaces
long_string = long_string.strip()

partitioned_string = partition(long_string)
print(partitioned_string)
```

**Comment by kingsamurai123** [comment]
Thanks this helped a lot. There is one another problem. If we re-run the script with the input as `utilities.json` or `minigames.json`, the description gets reformatted. So, do you think we can make boolean key value pair in plugins data with `formatted = true or false` for automation or go one by one for new plugins?

**Comment by rikkolovescats** [comment]
we shouldn't be overwritting .json files to include these added line breaks, the better way would be to call this partitioning function whenever we open up a plugin window, so it calculates and displays this newly partitioned string on the fly

**Comment by kingsamurai123** [comment]
Thought the same, but in that case there are two plugins already formatted `pro_unlocker` and `bomb_radius_visualizer` . Do you think we can remove the format from these plugins in the json file? If not, these descriptions may get extra new lines.

**Comment by rikkolovescats** [comment]
yep, will have to remove existing line breaks in plugin descriptions when merging this change

**Comment by kingsamurai123** [comment]
Found the description line in plugin manager:
https://github.com/bombsquad-community/plugin-manager/blob/2c8b17629b73f6d61400a06ed27785ab08066427/plugin_manager.py#L840
Added the partition function into this class:
https://github.com/bombsquad-community/plugin-manager/blob/2c8b17629b73f6d61400a06ed27785ab08066427/plugin_manager.py#L783
I have added the `plugin_manager.py` to my workspace in ballistica.net and opened bombsquad in my mobile. No matter how I try to call the `partition` function, the description, source and install gets removed from the text window. Can you help me with this?

**Comment by rikkolovescats** [comment]
can u make a PR? will make it easier for me to review

**Comment by kingsamurai123** [comment]
Tried calling the function using partition(<description>), <description>.partition and more. But nothing seems to be working. Maybe I am calling the function wrong?

**Comment by kingsamurai123** [comment]
Hi @rikkolovescats , 
I guess this issue is resolved with the above PR. Can you have a look at it?

**Comment by rikkolovescats** [comment]
closed by #97
